python/py_helpers.py

```diff
@@ -188,3 +188,78 @@ def _find_if_bodies(tree):
             return [tree.body] + [tree.orelse]
 
         return [Node(ast.Module(body, [])) for body in _find_if_bodies(self.tree)]
+
+    def find_whiles(self):
+        return self._find_all(ast.While)
+
+    def find_while_conditions(self):
+        def _find_while_conditions(tree):
+            if not isinstance(tree, ast.While):
+                return []
+            test = tree.test
+            if self.tree.orelse == []:
+                return [test]
+            if isinstance(tree.orelse[0], ast.While):
+                return [test] + _find_while_conditions(tree.orelse[0])
+
+            return [test, None]
+
+        return [Node(test) for test in _find_while_conditions(self.tree)]
+
+    def find_while_bodies(self):
+        def _find_while_bodies(tree):
+            if not isinstance(tree, ast.While):
+                return []
+            if self.tree.orelse == []:
+                return [tree.body]
+            if isinstance(tree.orelse[0], ast.While):
+                return [tree.body] + _find_while_bodies(tree.orelse[0])
+
+            return [tree.body] + [tree.orelse]
+
+        return [Node(ast.Module(body, [])) for body in _find_while_bodies(self.tree)]
+
+    def find_for_loops(self):
+        return self._find_all(ast.For)
+
+    def find_for_vars(self):
+        if not isinstance(self.tree, ast.For):
+            return Node()
+        return Node(self.tree.target)
+
+    def find_for_iter(self):        
+        if not isinstance(self.tree, ast.For):
+            return Node()
+        return Node(self.tree.iter)
+
+    def find_for_bodies(self):
+        def _find_for_bodies(tree):
+            if not isinstance(tree, ast.For):
+                return []
+            if self.tree.orelse == []:
+                return [tree.body]
+            if isinstance(tree.orelse[0], ast.For):
+                return [tree.body] + _find_for_bodies(tree.orelse[0])
+
+            return [tree.body] + [tree.orelse]
+
+        return [Node(ast.Module(body, [])) for body in _find_for_bodies(self.tree)]
+    
+    def find_if(self, if_str):
+        if_list = self._find_all(ast.If)
+        for if_statement in if_list:
+            if if_statement.find_conditions()[0].is_equivalent(if_str):                
+                return if_statement
```

Comment:
```suggestion
                return if_statement
        return Node()
```
Implicitly returning `None` breaks the chain and continued chaining gives you `AttributeError: 'NoneType' object has no attribute...` errors.